core/src/main/java/org/apache/accumulo/core/fate/zookeeper/MetaFateStore.java

```diff
@@ -301,15 +282,31 @@ public void push(Repo<T> repo) throws StackOverflowException {
 
       String txpath = getTXPath(fateId);
       try {
-        String top = findTop(txpath);
-        if (top != null && Long.parseLong(top.split("_")[1]) > 100) {
+        AtomicBoolean validDequeSize = new AtomicBoolean(false);
+
+        zk.mutateExisting(txpath, currSerFateData -> {
+          FateData<T> fateData = new FateData<>(currSerFateData);
+          Preconditions.checkState(REQ_PUSH_STATUS.contains(fateData.status),
+              "Tried to push to the repo stack for %s when the transaction status is %s", fateId,
+              fateData.status);
+          var repoDeque = fateData.repoDeque;
+          validDequeSize.set(repoDeque.size() < MAX_REPOS);
+
+          if (validDequeSize.get()) {
+            repoDeque.push(repo);
+            return fateData.serialize();
+          } else {
+            // This will not change the value and will return null
+            return null;
+          }
+        });
+
+        if (!validDequeSize.get()) {
           throw new StackOverflowException("Repo stack size too large");
         }
-
-        zk.putPersistentSequential(txpath + "/repo_", serialize(repo));
       } catch (StackOverflowException soe) {
```

Comment:
This was preexisting code.  Not sure why this is catching and rethrowing the exception.

Reply:
Yeah, maybe there was more code here at some point. Deleted the catch and rethrow